tests/unit/test_verification.cpp

```diff
@@ -129,6 +149,42 @@ TEST_F(VerificationTest, testFutureCertValidationFails)
     ASSERT_THROW(_root1_future->verify(trustStore, intermediateCAs), MoCOCrWException);
 }
 
+TEST_F(VerificationTest, testFutureCertValidationFailsWithTimeSetToNow)
+{
+    VerificationContext ctx;
+    ctx.addTrustedCertificate(*_root1.get())
+       .setVerificationCheckTime(Asn1Time::now());
+
+    ASSERT_THROW(_root1_future->verify(ctx), MoCOCrWException);
+}
+
+TEST_F(VerificationTest, testFutureCertValidationSucceedsWithTimeSetAccordingly)
+{
+    VerificationContext ctx;
+    ctx.addTrustedCertificate(*_root1_future.get())
+       .setVerificationCheckTime(_root1_future->getNotBeforeAsn1() + Asn1Time::Seconds(1));
+
+    _root1_future->verify(ctx);
+}
+
+TEST_F(VerificationTest, testCertValidationFailsWithTimeInFuture)
```

Comment:
@MAltgelt , tests content is fine, but names are a bit confusing. Wouldn't testCertValidationFailWhenCertificateIsNotYetValid () better reflect what we actually testing? (same for other added tests)

Reply:
Yes, but that name might also refer to a certificate that only becomes valid in the future. It doesn't imply that the time for the verification check is set.
Could we agree on testCertValidationFailWhenCertificateIsNotYetValidAtVerificationTime and similar names?

Reply:
I think testCert**Verification**FailWhenCertificateIsNotYetValidAt**Verification**Time() is an overkill. To me, it is obvious that we test "verification" at "verification" time. I do not have strong feelings about it, so if you disagree, let's leave it unchanged.

Reply:
I disagree with your proposed nomenclature, I think, but I haven't got strong feelings about the test names either, so should someone else request a change here, I'll go along.